paymentsheet/src/main/java/com/stripe/android/link/DismissalCoordinator.kt

```diff
@@ -0,0 +1,32 @@
+package com.stripe.android.link
+
+import javax.inject.Inject
+
+internal interface DismissalCoordinator {
```

Comment:
Introducing this new coordinator to communicate from the screen ViewModels to the activity ViewModel whether or not the bottom sheet can currently be dismissed, based on any critical network requests happening.

Turned out that simply hard-coding `confirmValueChange` in fe4e15c30b548edbf8243931c86b6a44ba7c5b54 didn't work out quite right, since it effectively turned off the dismissal animation.

Reply:
This is a great approach IMO. Let's add docs explaining this to the class? 

Also maybe `LinkDismissalCoordinator` to be more concrete / avoid confusion with PaymentSheet?